administrator/components/com_associations/models/fields/associatedcomponent.php

```diff
@@ -64,40 +64,46 @@ protected function getGroups()
 			'com_users',
 		);
 
+		$user       = JFactory::getUser();
+		$userId     = $user->id;
+
 		// Get all admin components.
 		foreach (glob(JPATH_ADMINISTRATOR . '/components/*', GLOB_NOSORT | GLOB_ONLYDIR) as $componentAdminPath)
 		{
 			$component           = basename($componentAdminPath);
 			$componentModelsPath = $componentAdminPath . '/models';
 
-			// Only components that exist also in the site client, aren't in the excluded components array and have models.
-			if (!is_dir($componentModelsPath) || in_array($component, $excludeComponents))
-			{
-				continue;
-			}
-
-			// Check if component uses associations, by checking is models.
-			foreach (glob($componentModelsPath . '/*.php', GLOB_NOSORT) as $modelFile)
+			if ($user->authorise('core.manage', $component))
 			{
```

Comment:
Maybe instead of
```php
if (!is_dir($componentModelsPath) || in_array($component, $excludeComponents))
```

you do ?
```php
if (!is_dir($componentModelsPath) || in_array($component, $excludeComponents) || !$user->authorise('core.manage', $component))
```